DependencyInjection/ClientFactory.php

```diff
@@ -331,4 +354,39 @@ protected function setupPaths(Client $client, $strip, $project, $root)
             }
         }
     }
+
+    /**
+     * Setup session tracking.
+     *
+     * @param \Bugsnag\Client $client
+     * @param string|null     $endpoint
+     */
+    protected function setupSessionTracking(Client $client, $endpoint = null)
+    {
+        if (class_exists(\Symfony\Component\Cache\Adapter\FilesystemAdapter::class)) {
+            $client->setAutoCaptureSessions(true);
+            if (!is_null($endpoint)) {
+                $client->setSessionEndpoint($endpoint);
+            }
+            $sessionTracker = $client->getSessionTracker();
+
+            $cache = new \Symfony\Component\Cache\Adapter\FilesystemAdapter();
+
+            $genericStorage = function ($key, $value = null) use ($cache) {
+                if (is_null($value)) {
+                    if ($cache->hasItem($key)) {
+                        return $cache->getItem($key)->get();
+                    } else {
+                        return;
```

Comment:
is this needed?

Reply:
```php
if (is_null($value) && $cache->hasItem($key)) {
    return $cache->getItem($key)->get();
 } else {
    $item = $cache->getItem($key);
    $item->set($value);
    $cache->save($item);
}

Reply:
Though, there is a race condition here anywhere, and two cache calls instead of one.

Reply:
Yeah, needs syncing of some sort.

Reply:
Just do the get by itself, right?